internal/storage/v2/elasticsearch/tracestore/to_dbmodel.go

```diff
@@ -29,7 +30,12 @@ const (
 
 // ToDBModel translates internal trace data into the DB Spans.
 // Returns slice of translated DB Spans and error if translation failed.
-func ToDBModel(td ptrace.Traces) []dbmodel.Span {
+func ToDBModel(td ptrace.Traces, allTagsAsObject bool, tagKeysAsFields []string, tagDotReplacement string) []dbmodel.Span {
+	tags := map[string]bool{}
+	for _, k := range tagKeysAsFields {
+		tags[k] = true
+	}
+	toDB := newToDBModel(allTagsAsObject, tags, tagDotReplacement)
 	resourceSpans := td.ResourceSpans()
 
 	if resourceSpans.Len() == 0 {
```

Comment:
if you must do this, shouldn't it be done at the very beginning of the function? And the 2nd question - why must you do this check in the first place? Isn't zero slice going to produce the exact same result anyway? This looks like a shortcut that provides no benefits (and in prod it's very unlikely to have a zero-length input anyway, so it's not any sensible performance saver).